cmd/tools/grafana/grafana.go

```diff
@@ -945,26 +945,33 @@ grafana export --addr my.grafana.server:3000 --serverfolder server_folder --dire
 }
 
 func init() {
-	Cmd.AddCommand(importCmd, exportCmd)
-
-	Cmd.PersistentFlags().StringVar(&opts.config, "config", "./harvest.yml", "harvest config file path")
-	Cmd.PersistentFlags().StringVarP(&opts.addr, "addr", "a", "http://127.0.0.1:3000", "Address of Grafana server (IP, FQDN or hostname)")
-	Cmd.PersistentFlags().StringVarP(&opts.token, "token", "t", "", "API token issued by Grafana server for authentication")
-	Cmd.PersistentFlags().StringVarP(&opts.prefix, "prefix", "p", "", "Use global metric prefix in queries")
-	Cmd.PersistentFlags().StringVarP(&opts.datasource, "datasource", "s", grafanaDataSource, "Grafana datasource for the dashboards")
-	Cmd.PersistentFlags().BoolVarP(&opts.variable, "variable", "v", false, "Use datasource as variable, overrides: --datasource")
-	Cmd.PersistentFlags().BoolVarP(&opts.useHTTPS, "https", "S", false, "Use HTTPS")
-	Cmd.PersistentFlags().BoolVarP(&opts.overwrite, "overwrite", "o", false, "Overwrite existing dashboard with same title")
-	Cmd.PersistentFlags().BoolVarP(&opts.useInsecureTLS, "insecure", "k", false, "Allow insecure server connections when using SSL")
-	Cmd.PersistentFlags().StringVarP(&opts.serverfolder.name, "serverfolder", "f", "", "Grafana folder name for dashboards")
-	Cmd.PersistentFlags().StringVarP(&opts.dir, "directory", "d", "", "When importing, import dashboards from this local directory.\nWhen exporting, local directory to write dashboards to")
+	Cmd.AddCommand(importCmd, exportCmd, metricsCmd)
+	addFlags(importCmd, exportCmd)
 
 	importCmd.PersistentFlags().StringSliceVar(&opts.labels, "labels", nil,
 		"For each label, create a variable and add as chained query to other variables")
 	importCmd.PersistentFlags().BoolVar(&opts.addMultiSelect, "multi", true,
 		"Modify the dashboards to add multi-select dropdowns for each variable")
 	_ = importCmd.PersistentFlags().MarkHidden("multi")
 
-	_ = Cmd.MarkPersistentFlagRequired("serverfolder")
-	_ = Cmd.MarkPersistentFlagRequired("directory")
```

Comment:
yep, they were never used

Reply:
Figured out what is going on:

Adding the `metricsCmd` broke the import/export `PersistentPreRun` func. 😐 That's because the `metricsCmd` set the `directory` flag to the default value `grafana/dashboards` for nicer looking help. Creating the command in that way then causes `PersistentPreRun` to see a value set for directory thereby defeating its logic.

Yesterday, because of the default value in `metricsCmd`, it appeared the commands I removed were not used since `MarkPersistentFlagRequired` was called on a parent cmd and not import/export and of course `serverfolder` is not required for metrics.

## Before PR, CLI works like so:

```
bin/grafana import --addr http://10.193.49.74:3000 -t $TOKEN
# runs fine, we use default serverfolder and directory
```

```
bin/grafana import --addr http://10.193.49.74:3000 --serverfolder x --directory grafana/dashboards/
# works
```

```
bin/grafana import --addr http://10.193.49.74:3000 --serverfolder foo
Error: required flag(s) "directory" not set
```

```
 bin/grafana import --addr http://10.193.49.74:3000 --directory x
Error: required flag(s) "serverfolder" not set
```

## With PR, CLI works like so:

```
bin/grafana import --addr http://10.193.49.74:3000 -t $TOKEN
# runs fine, we use default serverfolder and directory
```

```
bin/grafana import --addr http://10.193.49.74:3000 --serverfolder x --directory grafana/dashboards/
# works
```

```
bin/grafana import --addr http://10.193.49.74:3000 --serverfolder foo
Error: required flag(s) "directory" not set
```

```
 bin/grafana import --addr http://10.193.49.74:3000 --directory x
Error: required flag(s) "serverfolder" not set
```